sdk/core/azure-core/src/main/java/com/azure/core/util/BinaryData.java

```diff
@@ -1405,6 +1409,66 @@ public Flux<ByteBuffer> toFluxByteBuffer() {
     }
 
+    /**
+     * Writes the contents of this {@link BinaryData} to the given {@link OutputStream}.
+     * <p>
+     * This method does not close the {@link OutputStream}.
+     * <p>
+     * The contents of this {@link BinaryData} will be written without buffering. If the underlying data source isn't
+     * {@link #isReplayable()}, after this method is called the {@link BinaryData} will be consumed and can't be read
+     * again. If it needs to be read again, use {@link #toReplayableBinaryData()} to create a replayable copy.
+     *
+     * @param outputStream The {@link OutputStream} to write the contents of this {@link BinaryData} to.
+     * @throws NullPointerException If {@code outputStream} is null.
+     * @throws IOException If an I/O error occurs.
+     */
+    public void writeTo(OutputStream outputStream) throws IOException {
+        Objects.requireNonNull(outputStream, "'outputStream' cannot be null.");
+
+        content.writeTo(outputStream);
+    }
+
+    /**
+     * Writes the contents of this {@link BinaryData} to the given {@link WritableByteChannel}.
+     * <p>
+     * This method does not close the {@link WritableByteChannel}.
+     * <p>
+     * The contents of this {@link BinaryData} will be written without buffering. If the underlying data source isn't
+     * {@link #isReplayable()}, after this method is called the {@link BinaryData} will be consumed and can't be read
+     * again. If it needs to be read again, use {@link #toReplayableBinaryData()} to create a replayable copy.
+     *
+     * @param channel The {@link WritableByteChannel} to write the contents of this {@link BinaryData} to.
+     * @throws NullPointerException If {@code channel} is null.
+     * @throws IOException If an I/O error occurs.
+     */
+    public void writeTo(WritableByteChannel channel) throws IOException {
+        Objects.requireNonNull(channel, "'channel' cannot be null.");
+
+        content.writeTo(channel);
+    }
+
+    /**
+     * Writes the contents of this {@link BinaryData} to the given {@link AsynchronousByteChannel}.
+     * <p>
+     * This method does not close the {@link AsynchronousByteChannel}.
+     * <p>
+     * The contents of this {@link BinaryData} will be written without buffering. If the underlying data source isn't
+     * {@link #isReplayable()}, after this method is called the {@link BinaryData} will be consumed and can't be read
+     * again. If it needs to be read again, use {@link #toReplayableBinaryDataAsync()} to create a replayable copy.
+     *
+     * @param channel The {@link AsynchronousByteChannel} to write the contents of this {@link BinaryData} to.
+     * @return A {@link Mono} the completes once content has been written or had an error writing.
+     * @throws NullPointerException If {@code channel} is null.
+     * @throws IOException If an I/O error occurs.
+     */
+    public Mono<Void> writeToAsync(AsynchronousByteChannel channel) throws IOException {
```

Comment:
We can just call this method `writeTo` and make it just an overload as the arg will let the caller know this is async. Also, this method shouldn't throw but return error through Mono.